tests/php/TemplatingTest.php

```diff
@@ -192,6 +192,16 @@ public function testTemplateWithIfElseBlockAndNontruthfulCondition_ElseIsDisplay
 		$this->assertSame( '<p><a>else</a></p>', $result );
 	}
 
+	public function testTemplateWithElseIfBlockAndTruthfulCondition_IfAndElseRemoved() {
+		$result = $this->createAndRender(
+			'<p><a v-if="variable">if</a><a v-else-if="other_variable">else if</a>' .
+				'<a v-else>else</a></p>',
+			[ 'variable' => false, 'other_variable' => true ]
+		);
+
+		$this->assertSame( '<p><a>else if</a></p>', $result );
+	}
+
```

Comment:
To test the concern I mentioned above, we can add something like this:

```php
	public function testTemplateWithElseIfBlockAndTruthfulIfAndElseIfCondition_IfElseRemoved() {
		$result = $this->createAndRender(
			'<p><a v-if="variable">if</a><a v-else-if="other_variable">else if</a>' .
				'<a v-else>else</a></p>',
			[ 'variable' => true, 'other_variable' => true ]
		);

		$this->assertSame( '<p><a>if</a></p>', $result );
	}
```

Basically, when the `if` condition is true, the `else-if` element should not be rendered, even if its condition is also true.